sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -873,89 +903,75 @@ export class MessageReceiver extends LinkEntity {
         return;
       }
 
-      // We should attempt to reopen only when the receiver(sdk) did not initiate the close
-      let shouldReopen = false;
-      if (receiverError && !wasCloseInitiated) {
-        const translatedError = translate(receiverError) as MessagingError;
-        if (translatedError.retryable) {
-          shouldReopen = true;
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is retryable. This is a candidate for re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        } else {
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is NOT retryable. Hence NOT re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        }
-      } else if (!wasCloseInitiated) {
-        shouldReopen = true;
+      const translatedError = receiverError ? translate(receiverError) : receiverError;
+
+      // We should only attempt to reopen if either no error was present,
+      // or the error is considered retryable.
+      const shouldReopen =
+        !translatedError ||
+        (isMessagingError(translatedError) ? translatedError!.retryable : false);
```

Comment:
The OR ensures that translatedError is available, so do we need the ! here?

```suggestion
        (isMessagingError(translatedError) ? translatedError.retryable : false);
```

Reply:
Linter complained then, I assumed it wasn't smart enough.
It didn't complain now :(